src/olympia/reviewers/forms.py

```diff
@@ -497,8 +514,54 @@ def clean(self):
                 raise ValidationError(
                     'Multiple policies selected with different cinder actions.'
                 )
+
+        if self.helper.actions.get(selected_action, {}).get('delayable'):
+            if 'delayed_rejection' not in self.data:
+                self.add_error(
+                    'delayed_rejection',
+                    forms.ValidationError('This field is required.'),
+                )
+            elif self.cleaned_data.get('delayed_rejection') and not self.data.get(
+                'delayed_rejection_date'
+            ):
+                # In case reviewer selected delayed rejection option and
+                # somehow cleared the date widget, raise an error.
+                self.add_error(
+                    'delayed_rejection_date',
+                    forms.ValidationError('This field is required.'),
+                )
+            elif (
+                selected_action == 'change_pending_rejection_multiple_versions'
+                and self.cleaned_data.get('delayed_rejection')
+                and self.cleaned_data.get('delayed_rejection_date')
+                and self.cleaned_data.get('versions')
+            ):
+                distinct_pending_rejection_dates = (
+                    self.cleaned_data['versions']
+                    .values_list('reviewerflags__pending_rejection')
+                    .distinct()
+                    .count()
+                )
+                if distinct_pending_rejection_dates > 1:
+                    self.add_error(
+                        'versions',
+                        forms.ValidationError(
```

Comment:
Why is this a requirement? I could imagine having 3 vesrions with different dates and my goal is to set them all to the same single date??

Reply:
It's a requirement from the issue. I think the main problem is that we have to notify the developer of the old date in the email (see copy you commented on above), which makes it cumbersome if there are multiple different ones. Operations would know more, but @wagnerand noted the main use case would be when dates are all the same anyway so this limitation wouldn't be too bad.

Reply:
Ok